### ISSUE TITLE
Regression testing for new compilation pipeline

### ISSUE THREAD

**Issue by radumereuta** [kframework/k-legacy#1411]
So I manged to parse all programs. Had to fix a few issues in the definitions first, and adapt them to accept the new syntax for the new parser, but in the end everything worked as expected.

Today int he K meeting it was suggested that the new parser should be the default from now on, since users will have to switch to it eventually, and it will help me/us with the development of the new parser, since it will be tested continuously.

I talked to @dwightguth about this but he isn't that eager to make the switch since it is something that will be inconvenient for the users and it mainly benefits us as the developers.

Here are the typical modifications I had to do:

```
syntax Id ::= Token{[A-Za-z\_][A-Za-z0-9\_]*}   [notInRules]
syntax Id ::= Token{[A-Za-z\_][A-Za-z0-9\_]*}   [notInRules,
                    regex("(?<![A-Za-z0-9\\_])[A-Za-z\\_][A-Za-z0-9\\_]*"), token, autoReject]
```

```
syntax Id ::= "main"
syntax Id ::= "main" [token]
```

Basically, most tokens now have to be tagged with the regex specific for the java-regex engine, since SDF is not compatible. There are no longer token sorts, but only token productions which have to always be tagged with `token`.

One disadvantage for the users will be that they now have to understand the transition system also when specifying syntax.

Please comment on this issue so we can involve all people interested in this: @dwightguth @grosu @cos @radumereuta

**Comment by dwightguth** [comment]
My main concern with this is that we will cause problems for our users if the exact same string parses one way when used in a definition, but parses as something else if used in a program. And this is a definite possibility as a result of the ways in which these two parsing approaches have slightly different semantics. I would prefer for us to find a solution for us to test the new parser alongside the old implementation, rather than propagating a potentially confusing breaking user change to the entire framework.

**Comment by dwightguth** [comment]
I am going to rename this issue so that we can understand that it refers to our desire to maintain regression tests for the new pipeline. I don't think we actually want to make the new parser be the default yet, but we do need some way of testing both on every commit.
